File: binary_search.py
x=list(map(int,input().split()))
n=x[0]
s=x[1]
a=list(map(int,input().split()))


def func(a,s):
	lo=0
	hi=len(a)-1
	

	while lo<=hi:
		mid=int((hi+lo)/2)
		if a[mid]==s:
			return mid
		elif a[mid]<s:
			lo=mid+1
		else:
			hi=mid-1
	
	return -1		
				
print(func(a,s))
				

File: binary_search.cpp
#include<bits/stdc++.h>
using namespace std;



int main()
{
   int n,x;
   cin>>n>>x;
   int a[n];
   for(int i=0;i<n;i++)
   	   cin>>a[i];

   	int lo=0;
   	int hi=n-1;

   	
    int ans=-1;
   	while(lo<=hi)
   	{
   	  int mid=(hi-lo)/2+lo;
   	  if(x==a[mid])
   	  {
          ans=mid;
          break;
   	  }
   	  else if(a[mid]<x)
   	  {
          lo=mid+1;
   	  }	
   	  else
      {
          hi=mid-1;
      }
   	}
   	cout<<ans;
return 0;
}